src/SDK/FactoryRegistry.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\SDK;
+
+use OpenTelemetry\SDK\Common\Export\TransportFactoryInterface;
+use OpenTelemetry\SDK\Metrics\MetricExporterFactoryInterface;
+use OpenTelemetry\SDK\Trace\SpanExporter\SpanExporterFactoryInterface;
+use RuntimeException;
+
+class FactoryRegistry
+{
+    private static array $spanExporterFactories = [];
+    private static array $transportFactories = [];
+    private static array $metricExporterFactories = [];
+
+    /**
+     * @param callable-object|class-string $factory Factory implementing TransportFactoryInterface
```

Comment:
```suggestion
     * @param callable-object|class-string<TransportFactoryInterface> $factory Factory implementing TransportFactoryInterface
```

`callable-object` doesn't seem to match current implementation.

Reply:
thanks, phan was unhappy with `callable` but I think what I really wanted was just `TransportFactoryInterface`...